file_catalog/pathfinder.py

```diff
@@ -0,0 +1,81 @@
+"""Utility functions for finding a given filepath in the FC."""
+
+import os
+from typing import Any, Optional
+
+from .schema.types import Metadata
+
+
+def _is_conflict(uuid: Optional[str], file_found: Metadata) -> bool:
+    # if no file was found, then no problem
+    if not file_found:
+        return False
+
+    # if no uuid was provided, then any match is a conflict
+    if not uuid:
+        return True
+
+    # if the file we got isn't the one we're trying to update, that's a conflict
+    return file_found["uuid"] != uuid
+
+
+async def _contains_existing_logicalname(
+    apihandler: Any, metadata: Metadata, uuid: Optional[str] = None
+) -> bool:
+    # if the user provided a logical_name
+    if "logical_name" in metadata:
+        # try to load a file by that logical_name
+        file_found = await apihandler.db.get_file(
+            {"logical_name": metadata["logical_name"]}
+        )
+        # if we got a file by that logical_name
+        if _is_conflict(uuid, file_found):
+            # then that logical_name belongs to another file (already exists)
+            apihandler.send_error(
+                409,
+                reason=f"Conflict with existing file (logical_name already exists `{metadata['logical_name']}`)",
```

Comment:
From a security perspective, returning the source metadata without sanitizing it in the reason isn't great.  It might be better to just say:
```python
reason="Conflict with existing file (logical_name already exists)"
```